saleor/graphql/order/mutations/draft_order_complete.py

```diff
@@ -90,6 +104,7 @@ def perform_mutation(  # type: ignore[override]
 
         country = get_order_country(order)
         validate_draft_order(order, country, manager)
+        cls.setup_voucher_customer(order)
```

Comment:
I think it should be in the transaction as we do not want to create `VoucherCustomer` when the mutation fails.

Reply:
You can also consider using `channel` variable from line `125`

Reply:
good point